src/Settings/Builders/Make.hs

```diff
@@ -9,8 +10,16 @@ makeBuilderArgs = do
     threads    <- shakeThreads <$> expr getShakeOptions
     gmpPath    <- expr gmpBuildPath
     libffiPath <- expr libffiBuildPath
+    ghcPath    <- expr $
+      (-/-) <$> topDirectory <*> builderPath (Ghc CompileHs Stage2)
```

Comment:
@alpmestan This could be the reason? You might want to move this into the `nofib` subexpression.

Reply:
Here is the error, which mentions `ghc-source-path`:
```
Build system error - indirect recursion detected:
  Key value 1:  OracleQ (KeyValue ("hadrian/cfg/system.config","host-os"))
  Key value 2:  hadrian/cfg/system.config
  Key value 3:  hadrian/cfg/system.config settings mk/config.h compiler/ghc.cabal
  Key value 4:  OracleQ (KeyValue ("hadrian/cfg/system.config","ghc-source-path"))
Rules may not be recursive
```

Reply:
I don't even need this anymore. Let me try getting rid of the changes to this file altogether.